src/py/aspen/test_infra/aws.py

```diff
@@ -0,0 +1,15 @@
+import boto3
+import pytest
+from moto import mock_s3
+
+
+@pytest.fixture(scope="session")
+def s3_resource():
+    with mock_s3():
+        s3_resource = boto3.resource("s3")
+        yield s3_resource
+
+
+@pytest.fixture(scope="session")
+def s3_client(s3_resource):
+    yield s3_resource.meta.client
```

Comment:
It doesn't seem this is required, as you can use the high-level s3_resource API to upload.

Reply:
unfortunately you do need this to upload, otherwise i get `AttributeError: 's3.ServiceResource' object has no attribute 'put_object'`

Reply:
i can call this in the test directly though from the mock_s3_resource, i actually don't think we need to split this into the resource and client

Reply:
you can do:

`mock_s3_resource.Bucket(s3_bucket).Object(s3_key).put(Body=data)`

Reply:
ohhh 🤦‍♀️ lol

Reply:
done!